docs/standard/native-interop/best-practices.md

```diff
@@ -32,14 +38,16 @@ The guidance in this section applies to all interop scenarios.
 
 ## String parameters
 
-When the CharSet is Unicode or the argument is explicitly marked as `[MarshalAs(UnmanagedType.LPWSTR)]` _and_ the string is passed by value (not `ref` or `out`), the string will be pinned and used directly by native code (rather than copied).
+When <xref:System.Runtime.InteropServices.LibraryImportAttribute.StringMarshalling> is defined as <xref:System.Runtime.InteropServices.StringMarshalling.Utf16> or the argument is explicitly marked as `[MarshalAs(UnmanagedType.LPWSTR)]` or <xref:System.Runtime.InteropServices.DllImportAttribute.CharSet> is Unicode _and_ the string is passed by value (not `ref` or `out`), the string will be pinned and used directly by native code (rather than copied).
```

Comment:
```suggestion
The string is pinned and used directly by native code (rather than copied) when:

- <xref:System.Runtime.InteropServices.LibraryImportAttribute.StringMarshalling> is defined as <xref:System.Runtime.InteropServices.StringMarshalling.Utf16>.
- The argument is explicitly marked as `[MarshalAs(UnmanagedType.LPWSTR)]`.
- <xref:System.Runtime.InteropServices.DllImportAttribute.CharSet> is Unicode and the string is passed by value (not `ref` or `out`).
```